processor/resourcedetectionprocessor/internal/system/system.go

```diff
@@ -31,21 +32,35 @@ const (
 	TypeStr = "system"
 )
 
+var hostnameSourcesMap = map[string]func(*Detector) (string, error){
+	"dns": getFQDN,
+	"os":  getHostname,
+}
+
 var _ internal.Detector = (*Detector)(nil)
 
 // Detector is a system metadata detector
 type Detector struct {
-	provider systemMetadata
-	logger   *zap.Logger
+	provider        systemMetadata
+	logger          *zap.Logger
+	hostnameSources []string
 }
 
 // NewDetector creates a new system metadata detector
-func NewDetector(p component.ProcessorCreateSettings, _ internal.DetectorConfig) (internal.Detector, error) {
-	return &Detector{provider: &systemMetadataImpl{}, logger: p.Logger}, nil
+func NewDetector(p component.ProcessorCreateSettings, dcfg internal.DetectorConfig) (internal.Detector, error) {
+	cfg := dcfg.(Config)
+	err := cfg.Validate()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This should be called on a `Validate` method of the top-level configuration (there is no method as of now, it should have this type signature https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/10b008641faa7300222ff608a6337a6a8398caa0/receiver/jaegerreceiver/config.go#L87 ). That way this can be validated without creating the detector (e.g. in a dry run).

Reply:
I have added the suggested changes. Please, review the same.